doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,382 @@
+Hop Validation Field
+    22 bit Message Authentication Code to authenticate the FABRID extension metadata field.
+    With this the receiving endhost can be sure that the packet has actually been processed by that AS.
+Path Validator
+    4 byte Path Validator. The sending endhost will compute the path validator and the
+    receiving endhost can then recompute the path validator to verify that the packet
+    has been sent over the correct path.
+
+Identifier and FABRID Option combined
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Both the Identifier and FABRID extension together could look like this::
+
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+                                    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+                                    |  OptType = 3  |  OptLen = 8   |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |R R R R R|                Timestamp                            |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                          Packet ID                            |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |   Padding     |    Padding    |  OptType = 4  |  OptLen = ?   |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | Enc PolicyID  |F|A|   Hop Validation Field                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | Enc PolicyID  |F|A|   Hop Validation Field                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |    ....       | | |               ....                        |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    | Enc PolicyID  |F|A|   Hop Validation Field                    |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |                       Path Validator                          |
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+Header fields computation
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. math::
+    \begin{align*}
+        &encryptedPolicyID = policyID \oplus AES.Encrypt(K_i, Identifier)[0]\\\\
+        &policyID = encryptedPolicyID \oplus AES.Encrypt(K_i, Identifier)[0]\\\\
+        &K_i = DRKey\,(AS\,A_i \rightarrow AS_0:Endhost)\,or\,(AS\,A_i \rightarrow AS_0)\\\\
+        &HVF_i = MAC_{K_i}(Identifier, ingress_i, egress_i, encryptedPolicyID_i,\\& srcAddrLen, srcHostAddr)[0:3] \oplus 0x3FFFFF\\\\
+        &HVFVerified_i = MAC_{K_i}(Identifier, ingress_i, egress_i, encryptedPolicyID_i,\\& srcAddrLen, srcHostAddr)[3:6] \oplus 0x3FFFFF\\\\
+    \end{align*}
+
+Data plane
+----------
+
+Processing at the router
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Whenever a FABRID enabled router receives a SCION packet, it has to figure out whether it should be processed as FABRID or not.
```

Comment:
Maybe say at the beginning already that if an AS does not support FABRID, it forwards the packet as normal SCION packet.

Reply:
Done